frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1419,13 +1422,19 @@ export const surveyLogic = kea<surveyLogicType>([
         defaultInterval: [
             (s) => [s.survey],
             (survey: Survey): IntervalType => {
-                const start = dayjs(survey.start_date || survey.created_at)
-                const end = survey.end_date ? dayjs(survey.end_date) : dayjs()
-                const diffInWeeks = end.diff(start, 'weeks')
+                const start = getSurveyStartDateForQuery(survey)
+                const end = getSurveyEndDateForQuery(survey)
+                const diffInDays = dayjs(end).diff(dayjs(start), 'days')
 
-                if (diffInWeeks <= 4) {
+                if (diffInDays < 2) {
+                    return 'hour'
+                }
+                // less than a month
+                if (diffInDays < 30) {
                     return 'day'
-                } else if (diffInWeeks <= 12) {
+                }
+                // more than a month, less than 3 months
+                if (diffInDays < 90) {
```

Comment:
whats the context of this change?

Reply:
for < 2 days of data, I thought grouping by hour makes more sense instead of grouping by day, as there's not enough data yet to making grouping by day relevant.

it doesn't really change for other cases

Reply:
I think this can be another PR so we can easily revert if users complain/get confused.
the PR title and description is only about normalizing the start date, but it's doing something else as well.